test.py

```diff
@@ -241,9 +245,9 @@ async def add_test(test_file) -> None:
                     await self.add_test(test_file)
                     self.pending_test_count += 1
 
-            for p in patterns:
```

Comment:
Won't this change the behavior?

If I have a test file named `my_test` and I do
`./test.py --mode dev my_test my_test`

previously it would run the test twice because of the `for p in patterns` loop, now it will run the test once.

If the change is intended, the commit message should mention it, because it's not a simple readability change. It's a behavior change.

Reply:
Reworded in https://github.com/scylladb/scylladb/commit/9e1eb6bbcec97e76abdbc7933dd1d1468811453a